### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by irinamihai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,35 +22,40 @@ The spec allows you to define:
...
+* **UpgradeCannotStart**
+  * In this state, the upgrade cannot start because one the following reasons:
+    * Blocking CRs are missing from the system
+    * Blocking CRs have not yet reached the **UpgradeCompleted** state
 * **UpgradeNotCompleted**
-  * In this state, the controller will enforce the policies following the remediation plan built in the **UpgradeNotStarted** state
-  * The policies of subsequent batches will start as soon as the current batch policies are all compliant unless the batch times out, in which case the controller will move on to the next batch. The value of this batch timeout is the **ClusterGroupUpgrade** timeout divided by the remediation plan number of batches.
-  * Within a batch, the platform upgrade will be enforced first. Once the platform upgrade policy is compliant, then the operator upgrades policies will be enforced
+  * In this state, the controller will make enforce copied of the inform *managedPolicies* policies and adds clusters to the corresponding placement rules following the remediation plan built in the **UpgradeNotStarted** state.
```
Comment (alosadagrande) [inline README.md:44]: the controller will make enforced copies

**Review by alosadagrande — CHANGES_REQUESTED** [review]
check that typo, the rest lgtm

File: README.md [hunk 2]
```diff
@@ -7,6 +7,13 @@ Cluster Group Upgrades operator uses the following CRDs:
 
 * **ClusterGroupUpgrade**
 
+and it contains the following controllers:
+
+* **clustergroupupgrade** that's doing both pre-caching and cluster upgrade
```
Comment (jc-rh) [inline README.md:12]:
```suggestion
* **clustergroupupgrade** that's doing both preparation steps like pre-caching / backup and the actual cluster upgrade
```

File: README.md [hunk 3]
```diff
@@ -15,35 +22,40 @@ The spec allows you to define:
...
+  * In this state, the **ClusterGroupUpgrade** CR has just been created and the *enable* field is set to *false*
+  * The controller will build a remediation plan based on the *clusters* list and with *enable* fields like:
+    * If *canaries* field is defined with a list of clusters, the first batch(es) of the remediation plan will contain those clusters
     * The number of remediation batches will be the length of *clusters* divided by *maxConcurrency*, each batch with a length of *maxConcurrency* containing the clusters following the *clusters* list ordering
-  * The admin can make changes to *clusters* and *remediationStrategy* only in this state, it will ignore them in others.
-  * The controller will transition to **UpgradeNotCompleted** state once the *remediationAction* field is set to *enforce*
+  * The admin can make changes to *clusters*, *managedPolicies* and *enable* only in this state, it will ignore them in others.
+  * The controller will transition to **UpgradeNotCompleted** state once the *enable* field is set to *enforce* or to **UpgradeCannotStart** if there are issues preventing the upgrade.
```
Comment (jc-rh) [inline README.md:38]:
```suggestion
  * The controller will transition to **UpgradeNotCompleted** state once the *enable* field is set to *true* or to **UpgradeCannotStart** if there are issues preventing the upgrade.
```

File: README.md [hunk 4]
```diff
@@ -15,35 +22,40 @@ The spec allows you to define:
...
 * **UpgradeTimedOut**
   * In this state, the controller will periodically check if all the policies for the **ClusterGroupUpgrade** are compliant, and in that case it will transition to **UpgradeCompleted**. This is to give a chance for upgrades to catch up, as they could be taking long to complete due to network, CPU or other issues but they are not really stuck and they can indeed complete
 * **UpgradeCompleted**
   * In this state, the upgrades of the clusters are complete
-  * If the *deleteObjects* field is set to **true**, the controller will delete the underlying RHACM policies. This is to avoid having RHACM Hub to continously check for compliance since the upgrade has been successful
+  * If the *action.afterCompletion.deleteObjects* field is set to **true** by default, the controller will delete the underlying RHACM policies. This is to avoid having RHACM Hub to continously check for compliance since the upgrade has been successful.
+* **PrecachingRequired**
+  * In this state we need to have the images pre-cached before starting the upgrade.
```
Comment (rcarrillocruz) [inline README.md:56]: I will leave to @vitus133 but I believe there are multiple precaching states?

https://github.com/openshift-kni/cluster-group-upgrades-operator/blob/main/controllers/precacheFsm.go#L31

File: README.md [hunk 5]
```diff
@@ -15,35 +22,40 @@ The spec allows you to define:
...
     * If the policies for the upgrade have not turned to compliant within the *timeout* value specified in the *remediationStrategy*
 * **UpgradeTimedOut**
   * In this state, the controller will periodically check if all the policies for the **ClusterGroupUpgrade** are compliant, and in that case it will transition to **UpgradeCompleted**. This is to give a chance for upgrades to catch up, as they could be taking long to complete due to network, CPU or other issues but they are not really stuck and they can indeed complete
 * **UpgradeCompleted**
   * In this state, the upgrades of the clusters are complete
-  * If the *deleteObjects* field is set to **true**, the controller will delete the underlying RHACM policies. This is to avoid having RHACM Hub to continously check for compliance since the upgrade has been successful
+  * If the *action.afterCompletion.deleteObjects* field is set to **true** by default, the controller will delete the underlying RHACM policies. This is to avoid having RHACM Hub to continously check for compliance since the upgrade has been successful.
```
Comment (rcarrillocruz) [inline README.md:53]: I think also deletes placementbindings and placementrules

File: README.md [hunk 6]
```diff
@@ -15,35 +22,40 @@ The spec allows you to define:
...
+* **PrecachingRequired**
```
Comment (jc-rh) [inline README.md:54]: Move this section up?

File: README.md (hunk 4 above)
Comment (irinamihai) [reply README.md:56]: We have a separate section for pre-caching linked below, I think that could be enough.

File: README.md (hunk 5 above)
Comment (irinamihai) [reply README.md:53]: I will change to "underlying RHACM objects"

File: README.md (hunk 6 above)
Comment (irinamihai) [reply README.md:54]: I'd rather leave it here and rephrase to
> If the *action.afterCompletion.deleteObjects* field is set to **true**, the controller will delete the underlying RHACM objects (policies, placement bindings, placement rules, managed cluster views) once the upgrade completes.

File: README.md (hunk 2 above)
Comment (irinamihai) [reply README.md:12]: I will skip "backup" as I want to have this cherry-picked to the release-4.10 branch.

Comment (jc-rh) [reply README.md]: Sure, I didn't know backup wasn't in 4.10

File: README.md (hunk 6 above)
Comment (jc-rh) [reply README.md:54]: Sorry, I actually meant the Precaching state section. It's probably better to have the states in actual order?

File: README.md [hunk 7]
```diff
@@ -15,35 +22,40 @@ The spec allows you to define:
...
 * **UpgradeCompleted**
   * In this state, the upgrades of the clusters are complete
-  * If the *deleteObjects* field is set to **true**, the controller will delete the underlying RHACM policies. This is to avoid having RHACM Hub to continously check for compliance since the upgrade has been successful
+  * If the *action.afterCompletion.deleteObjects* field is set to **true** (which is the default value), the controller will delete the underlying RHACM objects (policies, placement bindings, placement rules, managed cluster views) once the upgrade completes. This is to avoid having RHACM Hub to continously check for compliance since the upgrade has been successful.
+* **PrecachingRequired**
+  * In this state we need to have the images pre-cached before starting the upgrade.
 
+## Precaching
+Found [here](/docs/pre-cache)
```
Comment (jc-rh) [inline README.md:58]: @sabbir-47 is there a similar md file for backup?

File: README.md [hunk 8]
```diff
@@ -7,6 +7,13 @@ Cluster Group Upgrades operator uses the following CRDs:
 
 * **ClusterGroupUpgrade**
 
+and it contains the following controllers:
+
+* **clustergroupupgrade** that's doing both pre-caching and cluster upgrade
+* **managedclusterForCGU** used for initially deploying SNOs
```
[the remark this answers is not in the document]
Comment (Missxiaoguo) [reply README.md:13]: It's not captured in an upstream documentation but only explained in the code as comments.

File: README.md (hunk 4 above)
Comment (rcarrillocruz) [reply README.md:56]: Then remove line 54.
Either you add all precaching related states, or you don't, but having just one
and leaving the reader to figure out the rest is on other link doesn't make sense to me.

File: README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (jc-rh) [reply README.md:13]: Thanks @Missxiaoguo 
https://issues.redhat.com/browse/CNF-5043

**Pull request merged**